### PR TITLE
Refactor faudio submodule

### PR DESCRIPTION
**Pull request by p01arst0rm**
moved FAudio files into its own directory structure,
and fixed a bad header import.

### CODE REVIEW

**Comment by hcorion** [comment]
Looks good, since your already working on FAudio, do you mind hooking it up so we can use the system FAudio libraries?

**Comment by p01arst0rm** [comment]
should i do that on this PR, or separately?

**Comment by hcorion** [comment]
Let's do it in this PR

**Comment by Mew8000** [comment]
G

File: 3rdparty/FAudio/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+set(FAUDIO_TARGET 3rdparty_dummy_lib)
+if(USE_FAUDIO)
+
+	# FAudio depends on SDL2
+	find_package(SDL2)
```
Comment (hcorion) [inline 3rdparty/FAudio/CMakeLists.txt:5]: If you have system FAudio we don't need to check if the user has SDL2

**Comment by p01arst0rm** [comment]
@hcorion should be good now :)

**Comment by Nekotekina** [comment]
I think USE_SYSTEM_FAUDIO doesn't exist, it needs to be created.

**Comment by Megamouse** [comment]
Is there really a point in moving the module to a subfolder if the only thing that changes is the CmakeLists ?

File: 3rdparty/FAudio/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	else()
+		# FAudio depends on SDL2
+		find_package(SDL2)
+		if (NOT SDL2_FOUND OR SDL2_VERSION VERSION_LESS 2.0.9)
+			message(FATAL_ERROR "-- RPCS3: FAudio requires SDL 2.0.9 or newer.")
+		endif()
+
+		set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared library")
+		add_subdirectory(FAudio EXCLUDE_FROM_ALL)
+		target_compile_definitions(FAudio INTERFACE -DHAVE_FAUDIO)
```
Comment (hcorion) [inline 3rdparty/FAudio/CMakeLists.txt:15]: We need this in the system variant as well I think, otherwise I don't rpcs3 will actually compile with FAudio

**Comment by hcorion** [comment]
> Is there really a point in moving the module to a subfolder if the only thing that changes is the CMakeLists ?

Good point, the reorganization is not necessary in this case, especially since the FAudio part of the 3rdparty CMakeLists is only a few lines.

**Comment by Nekotekina** [comment]
What happened with this PR? Looks stalled.

**Comment by p01arst0rm** [comment]
> What happened with this PR? Looks stalled.

apologies; was taking a break from working :)

i'm not 100% sure what @hcorion wants to be done with this PR, or if they want
this submodule refactored, i'd need to know that first :)

**Comment by hcorion** [comment]
Hey, yeah, so I think we should keep it simple and just keep it in the 3rdparty/CMakeLists.txt if there's not too much going on in, like is the case for this dependency. I think we should keep the work to allow support for building with the system faudio and the header fix.

**Comment by p01arst0rm** [comment]
okay, ill move some stuff back :)

**Comment by p01arst0rm** [comment]
reverted changes; kept improvements. should be good to merge :)

**Pull request merged**